core/tests/Test/Dscp/Crypto/MerkleTree.hs

```diff
@@ -66,10 +83,14 @@ spec_merkleTree = describe "Merkle Tree Tests" $ do
                     withIndices
 
             shouldBeFoundIn    proof (index, value) = Just value == lookup index proof
-            shouldBeNotFoundIn proof (index, _    ) = Nothing    == lookup index proof
+            shouldBeNotFoundIn proof (index, _    ) = isNothing $ lookup index proof
 
         in  case proof' of
                 Nothing    -> True
                 Just proof ->
                        all (shouldBeFoundIn    proof) pairsToLook
                     && all (shouldBeNotFoundIn proof) pairsToFail
+
+    it "splits and merges proofs and its data correctly" $ property $
+        \(proof :: MerkleProof Int) ->
+            uncurry mergeProofAndData (separateProofAndData proof) === Just proof
```

Comment:
It's also interesting to see them called vice-versa - `separateProfAndData $ mergeProofAndData (proof, data)`
(unless `mergeProofAndData` also takes only what `separateProfAndData` produces).